lib/cgo/tests/check_coin.transactions.c

```diff
@@ -0,0 +1,268 @@
+  cr_assert( result == SKY_OK );
+  seckeys.data = &seckey; seckeys.len = 1; seckeys.cap = 1;
+  result = SKY_coin_Transaction_SignInputs(handle, seckeys);
+  cr_assert( result == SKY_OK );
+  cr_assert( pOutput->Coins % 1000000 != 0 );
+  result = SKY_coin_Transaction_Verify(handle);
+  cr_assert( result == SKY_OK );
+
+  //Valid
+  ptx = makeTransaction(&handle);
+  pOutput = ptx->Out.data;
+  pOutput->Coins = 10000000;
+  pOutput++;
+  pOutput->Coins = 1000000;
+  result = SKY_coin_Transaction_UpdateHeader(handle);
+  cr_assert( result == SKY_OK );
+  result = SKY_coin_Transaction_Verify(handle);
+  cr_assert( result == SKY_OK );
+}
+
+
+Test(coin_transaction, TestTransactionPushInput)
+{
+  int result;
+  Transaction__Handle handle;
+  coin__Transaction* ptx;
+  coin__UxOut ux;
+  ptx = makeEmptyTransaction(&handle);
+  makeUxOut( &ux );
+  cipher__SHA256 hash;
+  result = SKY_coin_UxOut_Hash( &ux, &hash );
+  cr_assert( result == SKY_OK );
+  GoUint16 r;
+  result = SKY_coin_Transaction_PushInput(handle, &hash, &r);
+  cr_assert( result == SKY_OK );
+  cr_assert( r == 0 );
+  cr_assert( ptx->In.len == 1 );
+  cipher__SHA256* pIn = ptx->In.data;
+  cr_assert( eq( u8[sizeof(cipher__SHA256)], hash, *pIn) );
+
+  GoUint16 MaxUint16 = 0xFFFF;
+  int len = ptx->In.len;
+  void* data = malloc(len * sizeof(cipher__SHA256));
+  cr_assert(data != NULL);
+  registerMemCleanup(data);
+  memcpy(data, ptx->In.data, len * sizeof(cipher__SHA256) );
+  result = SKY_coin_Transaction_ResetInputs(handle, MaxUint16 + len);
+  cr_assert( result == SKY_OK );
+  memcpy(ptx->In.data, data, len * sizeof(cipher__Sig));
+  freeRegisteredMemCleanup(data);
+  makeUxOut( &ux );
+  result = SKY_coin_UxOut_Hash( &ux, &hash );
+  cr_assert( result == SKY_OK );
+  result = SKY_coin_Transaction_PushInput(handle, &hash, &r);
+  cr_assert( result != SKY_OK );
+
+}
+
+
+Test(coin_transaction, TestTransactionPushOutput)
+{
+  int result;
+  Transaction__Handle handle;
+  coin__Transaction* ptx;
+  ptx = makeEmptyTransaction(&handle);
+
+  cipher__Address addr;
+  makeAddress( &addr );
+  result = SKY_coin_Transaction_PushOutput( handle, &addr, 100, 150 );
+  cr_assert( result == SKY_OK );
+  cr_assert( ptx->Out.len == 1 );
+  coin__TransactionOutput* pOutput = ptx->Out.data;
+  coin__TransactionOutput output;
+  memcpy(&output.Address, &addr, sizeof(cipher__Address));
+  output.Coins = 100;
+  output.Hours = 150;
+  cr_assert( eq( type(coin__TransactionOutput), output, *pOutput ) );
+  for(int i = 1; i < 20; i++){
+    makeAddress( &addr );
+    result = SKY_coin_Transaction_PushOutput( handle, &addr, i * 100, i * 50 );
+    cr_assert( result == SKY_OK );
+    cr_assert( ptx->Out.len == i + 1 );
+    pOutput = ptx->Out.data;
+    pOutput += i;
+    memcpy(&output.Address, &addr, sizeof(cipher__Address));
+    output.Coins = i * 100;
+    output.Hours = i * 50;
+    cr_assert( eq( type(coin__TransactionOutput), output, *pOutput ) );
+  }
+}
+
```

Comment:
`TestTransactionSignInputs` is missing . Please check that all tests in `develop` have been added